crates/rover-client/src/operations/graph/check_workflow/runner.rs

```diff
@@ -236,3 +256,38 @@ fn get_lint_response_from_result(
         None => None,
     }
 }
+
+fn get_custom_response_from_result(
+    task_status: Option<CheckWorkflowTaskStatus>,
+    target_url: Option<String>,
+    results: Option<GraphCheckWorkflowQueryGraphCheckWorkflowTasksOnCustomCheckTaskResult>,
+) -> Option<CustomCheckResponse> {
+    match results {
+        Some(result) => {
+            let mut violations = Vec::with_capacity(result.violations.len());
+            for violation in result.violations {
+                let start_line = if let Some(source_locations) = &violation.source_locations {
+                    if !source_locations.is_empty() {
+                        Some(source_locations[0].start.line)
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                };
+                violations.push(Violation {
+                    level: violation.level.to_string(),
+                    message: violation.message,
+                    start_line,
+                    rule: violation.rule,
+                })
+            }
```

Comment:
I wonder if we could simplify this a bit? This feels like a classic use case for map (i.e. we've got a vector of violations and we want to convert them into something else), is there a reason not to go down that route?

Reply:
Updating to use iter().map()